src/components/screens/vacation-requests-screen.tsx

```diff
@@ -39,15 +39,17 @@ const VacationRequestsScreen = () => {
   const [vacationRequests, setVacationRequests] = useAtom(
     adminMode ? allVacationRequestsAtom : vacationRequestsAtom
   );
+  const [originalData, setOriginalData] = useState(vacationRequests);
```

Comment:
If we change vacationRequests based on the changes of original data we need to be sure that all changes that we made in delete or update vacation request functions also happens with original data. Now it is not happening, so if i change vacation request and trigger filter, nothing will be shown but in develop changes are displayed.